File: Data Structures & Algorithms/subsets/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsets(nums) {
        const dfs = (acc, cur, i) => {
            if (i === nums.length) return [...acc, cur]

            acc = dfs(acc, [...cur, nums[i]], i + 1)
            acc = dfs(acc, [...cur], i + 1)
            return acc
        }

        return dfs([], [], 0)
    }
}

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def bsearch(l, r) -> int:
            m = (r + l) // 2
            v = nums[m]
            if l > r:
                return -1
            elif v == target:
                return m
            elif v < target:
                return bsearch(m + 1, r)
            else:
                return bsearch(l, m - 1)
        
        return bsearch(0, len(nums) - 1)